docs/api/makeRenderRestHook.md

```diff
@@ -0,0 +1,103 @@
+---
+title: makeRenderRestHook()
+---
+
+```typescript
+function makeRenderRestHook(makeProvider: ProviderType): RenderRestHookFunction;
+```
+
+`makeRenderRestHook()` is useful to test hooks that rely on the `rest-hooks`. It creates a renderRestHook()
+function that mirrors [react-hooks-testing-library](https://github.com/mpeyper/react-hooks-testing-library)'s [renderHook()](https://github.com/mpeyper/react-hooks-testing-library#renderhookcallback-options) but does so with a `<Suspense/>` boundary
+as well as in a `<Provider />` context.
+
+## Arguments
+
+### makeProvider
+
+```typescript
+type ProviderType = (
+  manager: NetworkManager,
+  subManager: SubscriptionManager<any>,
+) => React.ComponentType<{
+  children: React.ReactChild;
+}>;
+```
+
+Function to generate the provider used in `renderRestHook()`. The purpose of this is to unify construction of
+providers as they both are initialized in a very different fashion.
+
+- [makeRestProvider()](./makeRestProvider.md)
+- [makeExternalCacheProvider()](./makeExternalCacheProvider.md)
+
+## Returns
+
+```typescript
+type RenderRestHookFunction = {
+  <T>(callback: () => T, initialState?: State<unknown>): {
+    readonly result: {
```

Comment:
I feel like adding typing for the API of react-hooks-testing-library makes these docs more brittle

Reply:
But the alternative is forcing them to go to an external site which can change and is extra work to find the info. Whereas I just have to be conscious everytime I bump that dependency that i update this.

Certainly not ideal but best I can come up with.

Reply:
fair enough